int32 frequency
int32 sample_rate
int32 autogain
int32 gain
int32 ppm_error
